### PR TITLE
handle null value exceptions when arguments are missing or Null in ca…

### PR DESCRIPTION
**Pull request by Zhangxunmt**
…ling RAG pipeline

### Description
"null_pointer_exception" + error_code:500 error is thrown in the conversational search Process Response, when Empty fields on interactions creates a null pointer exception, or arguments are missing through calling RAG pipeline. Details in https://github.com/opensearch-project/ml-commons/issues/1997.

This PR adds a try catch block to catch all these errors in the Process Response in GenerativeQA, and provide a general message to customers with 400 error code based on the suggestions from the pen test results. 
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by HenryL27 — APPROVED** [review]
I feel like the appropriate fix is in the message index but this works for now

**Comment by Zhangxunmt** [comment]
> I feel like the appropriate fix is in the message index but this works for now

It may be possible that the messages are used in other app cases which input, response can be null, etc. So I didn't add that part in the message index. Sounds like this is a good fix for now.

**Review by jngz-es — APPROVED** [review]

File: search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/GenerativeQAResponseProcessor.java [hunk 1]
```diff
@@ -58,7 +59,8 @@
  */
 @Log4j2
 public class GenerativeQAResponseProcessor extends AbstractProcessor implements SearchResponseProcessor {
-
+    public static String IllegalArgumentMessage =
+        "Please check the provided generative_qa_parameters are complete and non-null(https://opensearch.org/docs/latest/search-plugins/conversational-search/#rag-pipeline). Messages in the memory can not have Null value for input and response";
```
Comment (dhrubo-os) [inline search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/GenerativeQAResponseProcessor.java:63]: I didn't clearly understand this message. What did you mean by complete here? Can we rephrase the message?

File: search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/GenerativeQAResponseProcessor.java [hunk 2]
```diff
@@ -148,37 +150,51 @@ public SearchResponse processResponse(SearchRequest request, SearchResponse resp
...
+                }
             }
-        }
 
-        return insertAnswer(response, answer, errorMessage, interactionId);
+            return insertAnswer(response, answer, errorMessage, interactionId);
+        } catch (NullPointerException nullPointerException) {
+            throw new IllegalArgumentException(IllegalArgumentMessage);
+        } catch (Exception e) {
+            throw new OpenSearchException("GenerativeQAResponseProcessor failed in precessing response");
```
Comment (dhrubo-os) [inline search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/GenerativeQAResponseProcessor.java:196]: [nit] processing

Is `OpenSearchException` 400 level exception?

**Pull request merged**